include/libdnf/advisory/advisory_collection.hpp

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright (C) 2021 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF_ADVISORY_ADVISORY_COLLECTION_HPP
+#define LIBDNF_ADVISORY_ADVISORY_COLLECTION_HPP
+
+#include "advisory.hpp"
+#include "advisory_module.hpp"
+#include "advisory_package.hpp"
+
+#include <vector>
+
+
+namespace libdnf::advisory {
+
+//TODO(amatej): add unit tests for AdvisoryCollection
+class AdvisoryCollection {
+public:
+    /// Get all AdvisoryPackages stored in this AdvisoryCollection
+    ///
+    /// @param output           std::vector of AdvisorPackages used as output.
+    ///                         This is much faster than returning new std::vector and later joining
+    ///                         them when collecting AdvisoryPackages from multiple collections.
+    /// @param with_filenames   Filenames of AdvisoryPackages are not always useful, this allows skipping them.
+    ///                         The filename is stored as a c string (not libsolv id) this incurs slowdown.
+    void get_packages(std::vector<AdvisoryPackage> & output, bool with_filenames = false);
```

Comment:
I am not sure if `output` as argument is a good idea in public API (in bindings). What about to make it private and public API method will return vector? I know that it was like that in dnf-4 but it was not API.

Reply:
I also suggest that the public method will not have argument `with_filenames`. I also suppose that the public method will return advisories always with with_filenames. We should consider without filenames as an internal performance optimization.

Reply:
I have moved these functions into `private:` section `AdvisoryCollection` with a TODO that we maybe want to hide them in `Impl`? 
Since they were used in the test I have also added the public versions which return the vector and there is no `with_filenames` argument.